From 41d8a7186f0a88046076c5ffc38dcc02ba055410 Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Tue, 11 Sep 2018 22:52:12 -0500
Subject: [PATCH 23/N] win32: work around stat/fstat defines

On Windows, we need to override the default stat()/fstat() functions
because we need them to use `off_t` rather than 32-bit integers.

So we also need to override the macros that try to concatenate the
respective prefix and the terms `stat` or `fstat`.

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 deps/uv/include/uv.h | 6 ++++
 1 file changed, 6 insertions(+)

diff --git a/deps/uv/include/uv.h b/deps/uv/include/uv.h
index 9794d996..96719b2f 100644
--- a/deps/uv/include/uv.h
+++ b/deps/uv/include/uv.h
@@ -1217,6 +1217,12 @@ UV_EXTERN int uv_fs_scandir(uv_loop_t* loop,
                             uv_fs_cb cb);
 UV_EXTERN int uv_fs_scandir_next(uv_fs_t* req,
                                  uv_dirent_t* ent);
+#ifdef V8_OS_WIN
+#define uv_fs__stati64 uv_fs_stat
+#define uv_fs__fstati64 uv_fs_fstat
+#define uv_fs__stat64 uv_fs_stat
+#define uv_fs__fstat64 uv_fs_fstat
+#endif
 UV_EXTERN int uv_fs_stat(uv_loop_t* loop,
                          uv_fs_t* req,
                          const char* path,
-- 
2.19.0.rc1.windows.1

